jvm/src/main/java/com/microsoft/bond/protocol/CompactBinaryProtocolWriter.java

```diff
@@ -0,0 +1,383 @@
+package com.microsoft.bond.protocol;
+
+import com.microsoft.bond.BondDataType;
+import com.microsoft.bond.Metadata;
+import com.microsoft.bond.ProtocolType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Stack;
+
+// TODO: add Compact Binary protocol format comment (either here and/or in the deserializer and/or in package.html)
```

Comment:
We can discuss this offline. It is redundant to have protocol documentation in both reader and writer. Java is different since you get one public class per compilation unit. C++ and C# writers and readers are in the same compilation unit and thus share the same comment. 

---
In reply to: [116114036](https://github.com/tstein/bond/pull/4#discussion_r116114036) [](ancestors = 116114036)